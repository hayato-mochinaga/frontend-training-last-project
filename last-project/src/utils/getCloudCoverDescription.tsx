import FlareIcon from '@mui/icons-material/Flare';
import WbSunnyIcon from '@mui/icons-material/WbSunny';
import CloudIcon from '@mui/icons-material/Cloud';
import UmbrellaIcon from '@mui/icons-material/Umbrella';

export const getCloudCoverDescription = (cloudCover: number, rain: number) => {
    if (rain > 0) return { description: "雨", icon: <UmbrellaIcon style={{ fill: '#ffffff', marginRight: '0.3rem' }} /> };
    if (cloudCover <= 10) return { description: "快晴", icon: <FlareIcon style={{ fill: '#ffffff', marginRight: '0.3rem' }} /> };
    if (cloudCover <= 80) return { description: "晴れ", icon: <WbSunnyIcon style={{ fill: '#ffffff', marginRight: '0.3rem' }} /> };
    return { description: "曇り", icon: <CloudIcon style={{ fill: '#ffffff', marginRight: '0.3rem' }} /> };
};
