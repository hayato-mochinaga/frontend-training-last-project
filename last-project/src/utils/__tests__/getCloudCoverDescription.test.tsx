import { getCloudCoverDescription } from '../getCloudCoverDescription';

describe('getCloudCoverDescription', () => {

    // 雨が降っている場合
    it('雨の値が0より大きい場合、「雨」とUmbrellaIconを返す', () => {
        const result = getCloudCoverDescription(0, 1);
        expect(result).toMatchSnapshot(); // スナップショットを使って比較
    });

    // 快晴の条件
    it('雲量が10以下で雨が降っていない場合、「快晴」とFlareIconを返す', () => {
        const result = getCloudCoverDescription(0, 0);
        expect(result).toMatchSnapshot(); // スナップショットを使って比較
    });

    // 晴れの条件
    it('雲量が11から80までで雨が降っていない場合、「晴れ」とWbSunnyIconを返す', () => {
        const result = getCloudCoverDescription(11, 0);
        expect(result).toMatchSnapshot(); // スナップショットを使って比較
    });

    // 曇りの条件
    it('雲量が80を超える場合、またはそれ以外の場合で雨が降っていない場合、「曇り」とCloudIconを返す', () => {
        const result = getCloudCoverDescription(81, 0);
        expect(result).toMatchSnapshot(); // スナップショットを使って比較
    });

});
