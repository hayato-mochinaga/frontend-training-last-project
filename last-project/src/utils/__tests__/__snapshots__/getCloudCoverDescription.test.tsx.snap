// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCloudCoverDescription 雨の値が0より大きい場合、「雨」とUmbrellaIconを返す 1`] = `
{
  "description": "雨",
  "icon": <Memo
    style={
      {
        "fill": "#ffffff",
        "marginRight": "0.3rem",
      }
    }
  />,
}
`;

exports[`getCloudCoverDescription 雲量が10以下で雨が降っていない場合、「快晴」とFlareIconを返す 1`] = `
{
  "description": "快晴",
  "icon": <Memo
    style={
      {
        "fill": "#ffffff",
        "marginRight": "0.3rem",
      }
    }
  />,
}
`;

exports[`getCloudCoverDescription 雲量が11から80までで雨が降っていない場合、「晴れ」とWbSunnyIconを返す 1`] = `
{
  "description": "晴れ",
  "icon": <Memo
    style={
      {
        "fill": "#ffffff",
        "marginRight": "0.3rem",
      }
    }
  />,
}
`;

exports[`getCloudCoverDescription 雲量が80を超える場合、またはそれ以外の場合で雨が降っていない場合、「曇り」とCloudIconを返す 1`] = `
{
  "description": "曇り",
  "icon": <Memo
    style={
      {
        "fill": "#ffffff",
        "marginRight": "0.3rem",
      }
    }
  />,
}
`;
